server/refreshhandlers.go

```diff
@@ -301,13 +305,6 @@ func (s *Server) handleRefreshToken(w http.ResponseWriter, r *http.Request, clie
 		return
 	}
 
-	// Giant Swarm's custom group name prefixing
```

Comment:
This is a weird one. It seems like something changed because the prefixing tests on refresh were failing:
```
    --- FAIL: TestOAuth2DeviceFlow/verify_OIDC_group_prefixing_on_refresh_token (0.01s)
        server_test.go:1822: verify OIDC group prefixing on refresh token: Groups changed after refresh. Got groups != want groups:  {
              Groups: [
            -  "mock:mock:authors",
            +  "mock:authors",
              ],
             }
```
Suddenly we got a double prefixing. removing this here causes the tests to pass. I have to try refreshing on a test installation to confirm we no longer need this.